### PR TITLE
Add nix as alternative to docker

### PR DESCRIPTION
**Pull request by Cloudef**
* Adds flake.nix and default.nix for use with the nix package manager
* Adds ci jobs that builds using both clang and gcc (lots of warnings btw)
* Some minor fixes to make the project build on newer clang and gcc
* Instructions in readme
* Add option to run the main-server easily as well
* Add support for env variable bypass for the steam check in loader for linux users
* Run ci workflow in pull requests
https://github.com/TLeonardUK/ds3os/issues/30

### CODE REVIEW

**Comment by Cloudef** [comment]
Demo
termtosvg_frw40971

File: Source/Loader/Utils/SteamUtils.cs [hunk 1]
```diff
@@ -91,6 +91,10 @@ public static string GetGameInstallPath(string FolderName)
 
         public static bool IsSteamRunningAndLoggedIn()
         {
+            if (Environment.GetEnvironmentVariable("YES_STEAM_IS_RUNNING") == "1")
```
Comment (TLeonardUK) [inline Source/Loader/Utils/SteamUtils.cs:94]: Can you remove this please, as its really just a hack, the actual code here needs to be fixed really.

I think the better solution here is probably just going to be iterating the active processes and see if steam is running.

**Comment by TLeonardUK** [comment]
Looks good to me, baring the envvar check, I don't personally use nix at all, but I'm happy to add it if it's of value to others.

> Adds ci jobs that builds using both clang and gcc (lots of warnings btw)
Yeh that doesn't suprise me, this project was build pretty much 100% on windows using msvc, linux was mostly an afterthought.

File: Source/Loader/Utils/SteamUtils.cs (hunk 1 above)
Comment (Cloudef) [reply Source/Loader/Utils/SteamUtils.cs:94]: That wont work either as in wine / proton you won't see steam, it runs natively on linux side. What you could do however is load steamapi and check if steam is running through that as valve proxies libsteamapi between wine <-> native

Comment (Cloudef) [reply Source/Loader/Utils/SteamUtils.cs:94]: Simpler solution might be to check this env variable :thinking:
https://github.com/ValveSoftware/Proton/blob/proton_9.0/steam_helper/steam.cpp#L1624

Comment (TLeonardUK) [reply Source/Loader/Utils/SteamUtils.cs:94]: Ok. Well I'll merge the pull request, can follow up on this later.

**Pull request merged**